.github/workflows/demo-test.yml

```diff
@@ -1,12 +1,32 @@
 name: Deploy Tanssi Ecosystem Demo
 
 on:
+  push:
+    branches:
+      - staging
+      - main
   pull_request:
   workflow_dispatch:
 
 jobs:
+  detect-changes:
```

Comment:
I don't think we need an extra run here; use the [path filter](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore):
```
  pull_request:
    paths:
      - '**/*.sol'
```